spec/models/manageiq/providers/openstack/network_manager/cloud_subnet_refresh_spec.rb

```diff
@@ -0,0 +1,117 @@
+describe ManageIQ::Providers::Openstack::NetworkManager::CloudSubnet do
+  describe "refresh" do
+    before do
+      parent_ems = FactoryGirl.create(:ems_openstack_with_authentication)
+      @ems = parent_ems.network_manager
+      EvmSpecHelper.local_miq_server(:zone => Zone.seed)
+    end
+
+    it "removes subnet link to router if the interface is deleted" do
+      setup_mocked_collector
+
+      EmsRefresh.refresh(@ems)
+      expect(CloudSubnet.count).to eq(1)
+      expect(CloudNetwork.count).to eq(1)
+      expect(NetworkPort.count).to eq(1)
+      expect(NetworkRouter.count).to eq(1)
+      expect(CloudSubnet.first.network_router_id).to eq(NetworkRouter.first.id)
+
+      # simulate the interface between a router and subnet being removed on the OSP side.
+      allow_any_instance_of(ManageIQ::Providers::Openstack::Inventory::Collector::NetworkManager).to receive(:network_ports).and_return([])
+
+      EmsRefresh.refresh(@ems)
+      expect(CloudSubnet.count).to eq(1)
+      expect(CloudNetwork.count).to eq(1)
+      expect(NetworkPort.count).to eq(0)
+      expect(NetworkRouter.count).to eq(1)
+      expect(CloudSubnet.first.network_router_id).to be(nil)
+    end
+
+    it "targeted refresh shouldn't remove subnet link to router unless the interface is deleted" do
+      setup_mocked_collector
+      ::Settings.ems_refresh.openstack_network.allow_targeted_refresh = true
```

Comment:
btw. there is a helper for stubbing settings

https://github.com/Ladas/manageiq-providers-amazon/blob/7b1a7226dea0566fd17d3b06271782e213bf4969/spec/models/manageiq/providers/amazon/aws_refresher_spec_common.rb#L47

Does the direct assignment work?

Reply:
hm, the ```::Settings.ems_refresh.openstack_network.allow_targeted_refresh = false``` might bite us, since specs run in parallel in CI. So we should use the stub that is isolated to 1 spec

Reply:
The direct assignment does work, but you're right that there's risk of cross contamination since the tests are in parallel. I'll adjust to use the settings stub.

Reply:
right, probably do another cleanup PR, since it's in all specs